models/migrations/v29.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/go-xorm/xorm"
+)
+
+func changeMirrorIntervalToBigInt(x *xorm.Engine) (err error) {
+	type Repository struct {
+		ID      int64
+		OwnerID int64
+		Name    string
+	}
+	type Mirror struct {
+		ID       int64       `xorm:"pk autoincr"`
+		RepoID   int64       `xorm:"INDEX"`
+		Repo     *Repository `xorm:"-"`
+		Interval time.Duration
+	}
+
+	if setting.UseSQLite3 {
+		return nil
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	_, err = x.Query("ALTER TABLE mirror MODIFY `interval` BIGINT;")
```

Comment:
This might be better?
`if engine.Dialect().DataSourceName() == "mysql" { engine.Exec("ALTER TABLE mirror MODIFY interval BIGINT"); }`